### ISSUE TITLE
`jinja-variable-lower-case` wrongly detects escaped strings as variables

### ISSUE THREAD

**Issue by deadnews** [aristanetworks/j2lint#88]
## Description

The `jinja-variable-lower-case` rule in `j2lint` is incorrectly identifies escaped strings as variables.
According to the [Jinja2 documentation](https://jinja.palletsprojects.com/en/3.1.x/templates/#escaping), these strings should be treated as literal strings, not variables.

## Steps to Reproduce

```jinja2
{# test.yml.j2 #}
template: '{{ '{{ ToLower .Value }}' }}'
```

```console
$ j2lint test.yml.j2
test.yml.j2
├── test.yml.j2:1 All variables should use lower case: '{{ variable }}': ToLower (jinja-variable-lower-case)
└── test.yml.j2:1 All variables should use lower case: '{{ variable }}': Value (jinja-variable-lower-case)

Jinja2 linting finished with 2 error(s) and 0 warning(s)
```

## Context

```console
$ j2lint --version                                               
Jinja2-Linter Version v1.1.0
```

**Comment by gmuloc** [comment]
Thanks for opening the issue - same answer as here: https://github.com/aristanetworks/j2lint/issues/85#issuecomment-1874093145

**Comment by cdtomkins** [comment]
This will be closed by https://github.com/aristanetworks/j2lint/pull/132
